superset/dashboards/filters.py

```diff
@@ -74,12 +75,13 @@ def apply(self, query: Query, value: Any) -> Query:
 
         datasource_perms = security_manager.user_view_menu_names("datasource_access")
         schema_perms = security_manager.user_view_menu_names("schema_access")
-        published_dash_query = (
+        datesource_perm_query = (
             db.session.query(Dashboard.id)
             .join(Dashboard.slices)
             .filter(
                 and_(
-                    Dashboard.published == True,  # pylint: disable=singleton-comparison
+                    Dashboard.published.is_(True),
+                    ~Dashboard.roles.any(),
```

Comment:
This line is a little obscure. Maybe add a comment explaining what it does?

```suggestion
                    # has no role-based access specified
                    ~Dashboard.roles.any(),
```